src/components/accession2/EditLocationModal.tsx

```diff
@@ -0,0 +1,115 @@
+  const [storageLocationSelected, setStorageLocationSelected] = useState<StorageLocationDetails>();
+
+  const newRecord = {
+    facilityId: 0,
+    storageLocation: '',
+  };
+  const [record, , onChange] = useForm(newRecord);
+
+  useEffect(() => {
+    const setLocations = async () => {
+      if (record.facilityId) {
+        const response = await getLocations(record.facilityId);
+        if (response) {
+          setStorageLocations(response);
+        } else {
+          setStorageLocations([]);
+        }
+      }
+    };
+    setLocations();
+  }, [record.facilityId]);
+
+  const saveLocation = async () => {
+    if (accession) {
+      const response = await updateAccession2({
+        ...accession,
+        ...record,
+        storageCondition: storageLocationSelected?.storageCondition,
+        storageLocation: storageLocationSelected?.storageLocation,
+      });
+      if (response.requestSucceeded) {
+        reload();
+      }
+      onClose();
+    }
+  };
+
+  return (
+    <DialogBox
+      onClose={onClose}
+      open={open}
+      title={strings.LOCATION}
+      size='large'
+      middleButtons={[
+        <Button label={strings.CANCEL} type='passive' onClick={onClose} priority='secondary' key='button-1' />,
+        <Button onClick={saveLocation} label={strings.SAVE} key='button-2' />,
+      ]}
+    >
+      <Grid item xs={12} textAlign='left'>
+        <Grid item xs={12}>
+          <SelectT<Facility>
+            label={strings.LOCATION}
+            placeholder={strings.SELECT}
+            options={seedBanks}
+            onChange={(value: Facility) => onChange('facilityId', value.id)}
+            isEqual={(a: Facility, b: Facility) => a.id === b.id}
+            renderOption={(facility: Facility) => facility.name}
+            displayLabel={(facility: Facility) => facility?.name || ''}
+            selectedValue={seedBanks.find((sb) => sb.id === record.facilityId)}
+            toT={(name: string) => ({ name } as Facility)}
+            fullWidth={true}
+            readonly={true}
+          />
+        </Grid>
+        <Grid item xs={12} sx={{ marginTop: theme.spacing(2) }}>
+          <SelectT<StorageLocationDetails>
+            label={strings.SUB_LOCATION}
+            placeholder={strings.SELECT}
+            options={storageLocations}
+            onChange={(value: StorageLocationDetails) => setStorageLocationSelected(value)}
+            isEqual={(a: StorageLocationDetails, b: StorageLocationDetails) => a.storageLocation === b.storageLocation}
+            renderOption={(details: StorageLocationDetails) => details.storageLocation}
+            displayLabel={(details: StorageLocationDetails) => details?.storageLocation || ''}
+            selectedValue={storageLocationSelected}
+            toT={(name: string) => ({ storageLocation: name } as StorageLocationDetails)}
+            fullWidth={true}
+            readonly={true}
+          />
+        </Grid>
```

Comment:
feels like we could re-use this part? If so, let me update my review to make that part a separate component. We can pass in the grid-size props maybe. Waiting for Steve to merge his BE PR that supports storage location in the create accession request.

Reply:
> Yes, we can definitely reuse it. I can hold on this pr until we have the component to re use

Yes, thanks.

Reply:
I sent out a separate PR for reusable location/sub-location selector. https://github.com/terraware/terraware-web/pull/478

Reply:
Ignore earlier comment, made this change part of my original PR to avoid confusion. https://github.com/terraware/terraware-web/pull/471